library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/datasketches/cardinality/function/IterableToHllSketch.java

```diff
@@ -94,6 +99,8 @@ public HllSketch getHllSketch() {
     }
 
     public void setHllSketch(final HllSketch hllSketch) {
-        this.hllSketch = hllSketch;
+        if (hllSketch != null) {
+            this.hllSketch = hllSketch.copy();
+        }
```

Comment:
Previously this had no usages. I'm not sure that changing it to ignore if the sketch is already set is the best approach.

Reply:
I don't think that is what this change is doing, its checking if the newly supplied sketch is not null.

Reply:
You're right. This makes more sense, but I still question doing it. There's not much reason you'd want to set to `null`, but now if someone does want to do that it won't work and they will not know why unless they look at the code.

Reply:
I'm guessing its to avoid the null pointer exception as you can't call `copy()` on null as, a thought there could be an else statement for setting to null?